pkg/internal/api/db_user.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/huandu/go-sqlbuilder"
+
+	sqlutil "github.com/ClickHouse/terraform-provider-clickhouse/pkg/internal/sql"
+)
+
+type User struct {
+	Name               string `json:"name"`
+	PasswordSha256Hash string `json:"-"`
+}
+
+func (c *ClientImpl) CreateUser(ctx context.Context, serviceID string, user User) (*User, error) {
+	format := "CREATE USER `$?` IDENTIFIED WITH sha256_hash BY ${hash}"
+	args := []interface{}{
+		sqlbuilder.Raw(sqlutil.EscapeBacktick(user.Name)),
+		sqlbuilder.Named("hash", user.PasswordSha256Hash),
+	}
+
+	sb := sqlbuilder.Build(format, args...)
+
+	sql, args := sb.Build()
+
+	_, err := c.runQuery(ctx, serviceID, sql, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	createdUser, err := c.GetUser(ctx, serviceID, user.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	// We don't get the SHA back from clickhouse, so we assume the operation was successful and return it to the client.
+	createdUser.PasswordSha256Hash = user.PasswordSha256Hash
+
+	return createdUser, nil
+}
+
+func (c *ClientImpl) GetUser(ctx context.Context, serviceID string, name string) (*User, error) {
+	format := "SELECT name FROM system.users WHERE name = ${name} and storage = 'replicated'"
```

Comment:
Can you please add inline comment why storage is 'replicated'?

Reply:
done